### ISSUE TITLE
No public access to io::Error

### ISSUE THREAD

**Issue by albel727** [intgr/posix-acl#56]
There has to be a way to programmatically distinguish raw error codes as returned by the acl library calls. E.g. for my purposes it's important to know, if `acl_get_file()` fails with `ENOTSUP`, or `acl_set_file`() with `EROFS`, as opposed to other errors, but there's currently no straightforward way to do it.

Changing `IoErrorDetail`'s `err` field to `pub` visibility should be pretty uncontroversial, no? At the very least I'd like to have a `IoErrorDetail::raw_os_error() -> Option<i32>` method.

**Comment by intgr** [comment]
Hi. Yeah, distinguishing error types seems like a desirable thing.

Some types can already be distingushed using `ACLError.kind()`, e.g.

```rust
    use posix_acl::PosixACL;
    use std::io::ErrorKind;
    let err = PosixACL::read_acl("/tmp/this-file-does-not-exist").unwrap_err();
    assert_eq!(err.kind(), ErrorKind::NotFound);
```

Looks like [`ErrorKind::ReadOnlyFilesystem`](https://doc.rust-lang.org/stable/std/io/enum.ErrorKind.html#variant.ReadOnlyFilesystem) is currently gated behind the nightly-only `#![feature(io_error_more)]` flag, but there's no mapping for `ENOTSUP`. Not good enough.

I'll think about this a bit.

**Comment by intgr** [comment]
Have a look at #57, if that looks good to you, I can merge and release it. Example:

```rust
use posix_acl::PosixACL;
use std::io::ErrorKind;
let err = PosixACL::read_acl("/tmp/this-file-does-not-exist").unwrap_err();
assert_eq!(err.as_io_error().unwrap().raw_os_error().unwrap(), libc::ENOENT);
```

**Comment by albel727** [comment]
Yes, that would work very well for me, thank you!

**Comment by intgr** [comment]
I released posix-acl 1.1.0 just now.
